### ISSUE TITLE
dynamic import loaded twice in build mode

### ISSUE THREAD

**Issue by Flashantik** [nuxt/nuxt#19580]
### Environment

------------------------------
- Operating System: `Linux`
- Node Version:     `v16.14.2`
- Nuxt Version:     `3.2.3`
- Nitro Version:    `2.2.3`
- Package Manager:  `npm@7.17.0`
- Builder:          `vite`
- User Config:      `-`
- Runtime Modules:  `-`
- Build Modules:    `-`
------------------------------

### Reproduction

https://stackblitz.com/edit/github-nfcl83?file=pages%2Fthird.vue

```
npm run build
npm run preview
```

Open `/third` page and see DevTools -> Network, static_file loaded twice. 

### Describe the bug

dynamic imports in application hooks load file twice in build mode with SSR. In SPA and/or dev mode all works correct
<img width="789" alt="Screenshot 2023-03-10 at 12 41 05" src="https://user-images.githubusercontent.com/25429994/224253650-9a198f75-7857-41e2-a8b8-d6fee806eb0f.png">

**Comment by manniL** [comment]
Hey! Can't reproduce via StackBlitz.

<img width="1656" alt="image" src="https://user-images.githubusercontent.com/640208/224293062-b1587b53-5399-40a4-b73b-8bd15471cbf3.png">


Neither by navigating to the page ("client-side") nor hard reloading on `/third`.

Can you give more info? 🤔

**Comment by Flashantik** [comment]
> Hey! Can't reproduce via StackBlitz.
> 
> <img alt="image" width="1656" src="https://user-images.githubusercontent.com/640208/224293062-b1587b53-5399-40a4-b73b-8bd15471cbf3.png">
> 
> Neither by navigating to the page ("client-side") nor hard reloading on `/third`.
> 
> Can you give more info? 🤔

you must make `npm run build`  after that `npm run preview` open `/third `and click reload

5 min and i will upload the video

**Comment by Flashantik** [comment]
> > you must make npm run build after that npm run preview open /third and click reload
> 
> That's what I did. Maybe a browser-specific issue or based on your Anti Virus software installed? 🤔

I dont think so, because it's reproduce always in other 3 PC

**Comment by Flashantik** [comment]
> > you must make npm run build after that npm run preview open /third and click reload
> 
> That's what I did. Maybe a browser-specific issue or based on your Anti Virus software installed? 🤔


https://user-images.githubusercontent.com/25429994/224297110-f899d575-07e8-423e-8bf7-36fd368c6991.mp4



you also must disable filtrating of request because file with type javascript not displayed when the filter `JS` is on
<img width="318" alt="Screenshot 2023-03-10 at 15 51 57" src="https://user-images.githubusercontent.com/25429994/224297642-f4b507af-a223-4010-bebc-0ed01fe351ce.png">

<img width="299" alt="Screenshot 2023-03-10 at 15 52 00" src="https://user-images.githubusercontent.com/25429994/224297620-f9a6c612-f6d4-47cd-9520-5c7dfff4eaac.png">

**Comment by manniL** [comment]
Same result 
<img width="1627" alt="image" src="https://user-images.githubusercontent.com/640208/224304992-15be9bea-6ed7-4fe3-8369-5e9bc1c48982.png">

**Comment by danielroe** [comment]
Looking into this, I don't think this is an issue. The file isn't being downloaded; it's being fetched from the prefetch cache, exactly as desired:

![CleanShot 2023-03-20 at 13 06 09](https://user-images.githubusercontent.com/28706372/226347529-47fc42ea-24c3-4cc5-8fb9-c0206dadc025.png)

**Comment by Flashantik** [comment]
> Looking into this, I don't think this is an issue. The file isn't being downloaded twice; it's being fetched from the prefetch cache, exactly as desired:
> 
> ![CleanShot 2023-03-20 at 13 06 09](https://user-images.githubusercontent.com/28706372/226347529-47fc42ea-24c3-4cc5-8fb9-c0206dadc025.png)
No, this requests not related, and browser start fetching first request and tries to load the next file, and you have 2 active requests. After that if you reload the page yeah, script file will be received from cache.

**Comment by Flashantik** [comment]
> If your app hydrates and starts dynamically importing that file before it is finished prefetching, then it is expected behaviour that you will have two network requests.

Why another browsers  has only on network request?

**Comment by keupsonite** [comment]
<img width="1391" alt="Capture d’écran 2024-03-03 à 15 14 33" src="https://github.com/nuxt/nuxt/assets/2184898/dbaccc9e-8ed3-4c2e-804e-d3ea9b2eb9f6">
Same problem, but second request not received from cache
